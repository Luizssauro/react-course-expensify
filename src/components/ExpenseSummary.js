import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ expenseCount = 0, expensesTotal = 0 }) => {
	const expenseWord = expenseCount > 1 ? 'expenses' : 'expense';
	return (
		<div>
			<p>
				{expenseCount === 0
					? 'There is no expenses yet!'
					: `Viewing ${expenseCount} ${expenseWord} totalling ${numeral(
							expensesTotal / 100
					  ).format('$0,0.00')}`}
			</p>
		</div>
	);
};

const mapStateToProps = state => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);

	return {
		expenseCount: visibleExpenses.length,
		expensesTotal: selectExpensesTotal(visibleExpenses)
	};
};

export default connect(mapStateToProps)(ExpenseSummary);
