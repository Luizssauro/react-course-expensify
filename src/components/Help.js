import React from 'react';

const Help = (props) => (
  <div>
    This is the help page
  </div>
);

export default Help;