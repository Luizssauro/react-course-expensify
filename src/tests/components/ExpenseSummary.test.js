import React from 'react';
import { shallow } from 'enzyme';
import numeral from 'numeral';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

describe('<ExpenseListItem />', () => {
	test('should render ExpenseSummary whithout expenses', () => {
		const wrapper = shallow(<ExpenseSummary />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('p').text()).toMatch('There is no expenses yet!');
	});

	test('should render ExpenseSummary whith expenses', () => {
		const wrapper = shallow(
			<ExpenseSummary expenseCount={5} expensesTotal={5142563} />
		);
		expect(wrapper).toMatchSnapshot();
	});

	test('should render ExpenseSummary whith only one expense', () => {
		const wrapper = shallow(
			<ExpenseSummary expenseCount={1} expensesTotal={3333} />
		);
		expect(wrapper).toMatchSnapshot();
	});
});
