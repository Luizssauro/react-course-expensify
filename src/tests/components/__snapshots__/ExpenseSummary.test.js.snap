// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseListItem /> should render ExpenseSummary whith expenses 1`] = `
<div>
  <p>
    Viewing 5 expenses totalling $51,425.63
  </p>
</div>
`;

exports[`<ExpenseListItem /> should render ExpenseSummary whith only one expense 1`] = `
<div>
  <p>
    Viewing 1 expense totalling $33.33
  </p>
</div>
`;

exports[`<ExpenseListItem /> should render ExpenseSummary whithout expenses 1`] = `
<div>
  <p>
    There is no expenses yet!
  </p>
</div>
`;
